src/sentry/tagstore/snuba/backend.py

```diff
@@ -689,6 +689,11 @@ def get_group_ids_for_search_filter(
         # instead.
         raise NotImplementedError
 
+    def delete_tag_key(self, project_id, key):
+        from sentry import eventstream
+        eventstream_state = eventstream.start_delete_tag(project_id, key)
+        eventstream.end_delete_tag(eventstream_state)
+
 
 class SnubaCompatibilityTagStorage(SnubaTagStorage):
```

Comment:
It's unclear to me whether my `delete_tag_key` impl belongs in `SnubaCompatibilityTagStorage` or in the real one (where it is now).

Reply:
> I do wonder if the `eventstream` calls should just wrap the call site instead, and we should move the actual call to `delete_tag_key` into the compatibility backend as a noop based on the precedent from the other operations?

Good call.

Reply:
Although -- that works today (and hopefully forever) but we have this weird case like the other endpoints where `tagstore.delete_tag_key` could fire off an async task which itself needs to send the `end` message.

Buuut... we can fix that if we ever come to it? And hopefully we never have to? (Long live Snuba)

Reply:
OK, I moved it to the API call site, so it's not dependent on the backend.

I *did* however punt on passing down the `eventstream_state` and requiring the `tagstore.delete_tag_key` callee to handle sending the end message. This means for the existing v2 backend the `end_delete_tag` message will most likely be sent before the v2 Celery task finishes running.

That seemed like a lot of effort and ugly APIs to perfectly align the timestamp for a message that has nothing to do with v2 in practice.

Reply:
> I _did_ however punt on passing down the `eventstream_state` and requiring the `tagstore.delete_tag_key` callee to handle sending the end message. This means for the existing v2 backend the `end_delete_tag` message will most likely be sent before the v2 Celery task finishes running.

I think that's a reasonable compromise, and probably worth noting here as a comment?